applications/ShallowWaterApplication/custom_utilities/shallow_water_particle.h

```diff
@@ -0,0 +1,180 @@
+a  copy  of this  software  and  associated  documentation files  (the
+"Software"), to  deal in  the Software without  restriction, including
+without limitation  the rights to  use, copy, modify,  merge, publish,
+distribute,  sublicense and/or  sell copies  of the  Software,  and to
+permit persons to whom the Software  is furnished to do so, subject to
+the following condition:
+
+Distribution of this code for  any  commercial purpose  is permissible
+ONLY BY DIRECT ARRANGEMENT WITH THE COPYRIGHT OWNER.
+
+The  above  copyright  notice  and  this permission  notice  shall  be
+included in all copies or substantial portions of the Software.
+
+THE  SOFTWARE IS  PROVIDED  "AS  IS", WITHOUT  WARRANTY  OF ANY  KIND,
+EXPRESS OR  IMPLIED, INCLUDING  BUT NOT LIMITED  TO THE  WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT  SHALL THE AUTHORS OR COPYRIGHT HOLDERS  BE LIABLE FOR ANY
+CLAIM, DAMAGES OR  OTHER LIABILITY, WHETHER IN AN  ACTION OF CONTRACT,
+TORT  OR OTHERWISE, ARISING  FROM, OUT  OF OR  IN CONNECTION  WITH THE
+SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+==============================================================================
+*/
+
+//
+//   Project Name:        Kratos
+//   Last Modified by:    Miguel Masó Sotomayor
+//   Date:                September 19th 2017
+//   Revision:            2.1
+//
+
+
+#if !defined(KRATOS_SHALLOW_WATER_PARTICLE_H_INCLUDED )
+#define  KRATOS_SHALLOW_WATER_PARTICLE_H_INCLUDED
+
+
+
+// System includes
+#include <string>
+#include <iostream>
+#include <sstream>
+#include <cstddef>
+
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "containers/array_1d.h"
+#include "includes/serializer.h"
+#include "geometries/point.h"
+#include "includes/model_part.h"
+
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// PFEM Particle class.
+
+/*
+@see Geometry
+@see Node
+@see IntegrationPoint
+*/
+//template<std::size_t TDimension, class TDataType = double> //always size 3!
+class ShallowParticle : public Point
+{
+public: 
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of Point
+    typedef ShallowParticle* Pointer;
+    
+    typedef double TDataType; 
+
+    typedef array_1d<double,3> BaseType;
+
+    typedef Point Type;
+
+    typedef BaseType CoordinatesArrayType;
+
+    typedef std::size_t SizeType;
+
+    typedef std::size_t IndexType;
+
+    ///@}
+    ///@name Constants
+    ///@{
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    ShallowParticle(TDataType const& NewX, TDataType const& NewY, TDataType const& NewZ) : Point(NewX, NewY, NewZ)
+    {
+        this->ERASE_FLAG = true; //initializing as useless particle
```

Comment:
try to avoid all caps names for variables (we reserve them for Kratos::Variable objects)